x-pack/plugins/endpoint/server/routes/resolver/tree.ts

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { schema } from '@kbn/config-schema';
+import { RequestHandler, Logger } from 'kibana/server';
+import { Fetcher } from './utils/fetch';
+import { Tree } from './utils/tree';
+
+interface TreeQueryParams {
+  // the rough approximation of how many children you can request per process, per level
```

Comment:
This type, as well as `TreePathParams` can be calculated from `validateTree.params` and `validateTree.query` respectively using `TypeOf<typeof validateTree.params>` and `TypeOf<typeof validateTree.query>`.